base/steer/FairRun.cxx

```diff
@@ -79,14 +79,26 @@ void FairRun::AddTask(FairTask* t)
 {
   fTask->Add(t);
   fNTasks++;
+  StoreTaskNames(t);
+}
+//_____________________________________________________________________________
+void FairRun::StoreTaskNames(FairTask* t)
+{
+  if(t==NULL) return;
   fFileHeader->AddTaskClassName(t->ClassName());
+  if (t->GetListOfTasks()==NULL) return;
+  for(TObject* t2: *(t->GetListOfTasks()) )
+  {
+    StoreTaskNames((FairTask*)t2);
+  }
+  return;
 }
 //_____________________________________________________________________________
 void FairRun::SetTask(FairTask* t)
 {
   delete fTask;
   fTask = t;
-  fFileHeader->AddTaskClassName(t->ClassName());
+  StoreTaskNames(t);
```

Comment:
Since `fTask` is deleted, shouldn't you also clear the previously stored task names?

Reply:
This is the same logic as before. I concur, though to clear the list. However, everything that was inside before will be gone, too.

Reply:
True, maybe @MohammadAlTurany or @fuhlig1 have a comment here, if the previous behaviour should be preserved or if clearing the class names is the better way to go.